dbt/adapters/snowflake/connections.py

```diff
@@ -63,6 +72,73 @@
 }
 
 
+@cache
+def _private_key(
+    private_key: Optional[str] = None,
+    private_key_path: Optional[str] = None,
+    private_key_passphrase: Optional[str] = None,
+) -> Optional[bytes]:
+    """Get Snowflake private key by path, from a Base64 encoded DER bytestring or None."""
+
+    if private_key and private_key_path:
+        raise DbtConfigError("Cannot specify both `private_key`  and `private_key_path`")
+    elif private_key:
+        p_key = _private_key_from_string(private_key, private_key_passphrase)
+    elif private_key_path:
+        p_key = _private_key_from_file(private_key_path, private_key_passphrase)
+    else:
+        return None
+
+    return p_key.private_bytes(
+        encoding=serialization.Encoding.DER,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+
+
+@cache
```

Comment:
Nit: `cache` here is redundant since this method should only get called exactly once.

Reply:
Agreed, but I figured I might as well do it here too as it's part of the same logical collection.